packages/react/src/host/RemoteComponent.tsx

```diff
@@ -19,45 +31,76 @@ interface Props {
   __type__?: ComponentType;
 }
 
+const emptyObject = {};
+
 export const RemoteComponent = memo(
   ({receiver, component, controller}: Props) => {
     const Implementation = controller.get(component.type)!;
 
     const attached = useAttached(receiver, component);
-    const props = attached?.props;
 
-    useEffect(() => {
-      retain(props);
+    const props = useMemo(() => {
+      const props = attached?.props as any;
+      if (!props) return emptyObject;
 
-      return () => {
-        release(props);
-      };
-    }, [props]);
+      return Object.keys(props).reduce((acc, key) => {
```

Comment:
What benchmarking tool is this? Just curious.

Reply:
I just used the first thing that came up on google for me, https://jsbench.me, although do not recommend it because I created an account to share the actual URL for the test and it never sent me a confirmation email :P